scripts/data/client.py

```diff
@@ -102,124 +110,132 @@ def process_batch(job):
-        logger.info(f"{job} done, {gas_info}")
-        if not match:
-            logger.warning(f"{job}: not gas info found")
+    except Exception as e:
+        logger.error(f"Error processing batch {job}: {e}")
 
 
-# Producer function: Generates data and adds jobs to the queue
 def job_producer(job_gen):
     global current_weight
 
     try:
         for job, weight in job_gen:
-            # Wait until there is enough weight capacity to add the new block
             with weight_lock:
                 logger.debug(
                     f"Adding job: {job}, current total weight: {current_weight}..."
                 )
                 while (
-                    (current_weight + weight > MAX_WEIGHT_LIMIT)
-                    and current_weight != 0
-                    or job_queue.full()
-                ):
+                    current_weight + weight > MAX_WEIGHT_LIMIT and current_weight != 0
+                ) or job_queue.full():
+                    if shutdown_event.is_set():
+                        return
                     logger.debug("Producer is waiting for weight to be released.")
-                    weight_lock.wait()  # Wait for the condition to be met
+                    weight_lock.wait(timeout=1)
 
-                if (current_weight + weight > MAX_WEIGHT_LIMIT) and current_weight == 0:
+                if current_weight + weight > MAX_WEIGHT_LIMIT and current_weight == 0:
                     logger.warning(f"{job} over the weight limit: {MAX_WEIGHT_LIMIT}")
 
-                # Add the job to the queue and update the weight
                 job_queue.put((job, weight))
                 current_weight += weight
-                logger.debug(
-                    f"Produced job: {job}, current total weight: {current_weight}"
-                )
-
-                # Notify consumers that a new job is available
                 weight_lock.notify_all()
     finally:
         logger.debug("Producer is exiting...")
+        # Signal end of work to consumers
         for _ in range(THREAD_POOL_SIZE):
             job_queue.put(None)
-
         with weight_lock:
             weight_lock.notify_all()
 
-        logger.debug("Consumers notified")
 
-
-# Consumer function: Processes blocks from the queue
 def job_consumer(process_job):
     global current_weight
 
     while True:
         try:
-            logger.debug(
-                f"Consumer is waiting for a job. Queue length: {job_queue.qsize()}"
-            )
-            # Get a job from the queue
-            work_to_do = job_queue.get(block=True)
+            work_to_do = job_queue.get()
 
             if work_to_do is None:
-                logger.debug("No more work to do, consumer is exiting.")
                 job_queue.task_done()
                 break
 
-            (job, weight) = work_to_do
-
-            # Process the block
+            job, weight = work_to_do
             try:
```

Comment:
Why log is removed here?

Reply:
Resolved and properly handled in the latest pr